src/gmtset.c

```diff
@@ -65,25 +65,29 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GMTSET_CTRL *C) {	/* Dealloc
-	GMT_Message (API, GMT_TIME_NONE, "\n\tFor available PARAMETERS, see %s man page.\n\n", GMT_SETTINGS_FILE);
+	GMT_Usage (API, 0, "usage: %s [-C | -D[s|u] | -G<defaultsfile>] [-[" GMT_SHORTHAND_OPTIONS "]<value>] [%s] PARAMETER1 value1 PARAMETER2 value2 PARAMETER3 value3 ...\n", name, GMT_V_OPT);
+	GMT_Usage (API, 1, "Give pairs of PARAMETER value to change these settings.  For available PARAMETERS, see %s documentation.", GMT_SETTINGS_FILE);
 
 	if (level == GMT_SYNOPSIS) return (GMT_MODULE_SYNOPSIS);
 
 	GMT_Message (API, GMT_TIME_NONE, "\n  OPTIONAL ARGUMENTS:\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-C Convert GMT4 .gmtdefaults4 to a %s file.\n", GMT_SETTINGS_FILE);
-	GMT_Message (API, GMT_TIME_NONE, "\t   The original file is retained.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-D Modify the default settings based on the GMT system defaults.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   Append s to see the SI version of defaults.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   Append u to see the US version of defaults.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-G Set name of specific %s file to modify.\n", GMT_SETTINGS_FILE);
-	GMT_Message (API, GMT_TIME_NONE, "\t   [Default looks for file in current directory.  If not found,\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   it looks in the home directory, if not found it uses GMT defaults.]\n");
-	GMT_Message (API, GMT_TIME_NONE, "\n\tOnly settings that differ from the GMT SI system defaults are written\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   to the file %s in the current directory (under classic mode)\n", GMT_SETTINGS_FILE);
-	GMT_Message (API, GMT_TIME_NONE, "\t   or in the current session directory (under modern mode).\n");
-	GMT_Message (API, GMT_TIME_NONE, "\n\t-[" GMT_SHORTHAND_OPTIONS "]<value> (any of these options).\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   Set the expansion of any of these shorthand options.\n");
+	GMT_Usage (API, 1, "\n-C Convert GMT4 .gmtdefaults4 to a %s file. "
+		"The original file is retained.", GMT_SETTINGS_FILE);
+	GMT_Usage (API, 1, "\n-D[s|u]");
+	GMT_Usage (API, -2, "Modify the default settings based on the GMT system defaults. Optionally append a directive:");
+	GMT_Usage (API, 3, "s: Use the SI version of defaults.");
+	GMT_Usage (API, 3, "u: Use the US version of defaults.");
+	GMT_Usage (API, 1, "\n-G<defaultsfile>");
+	GMT_Usage (API, -2, "Set name of specific %s file to modify "
+		"[Default looks for file in current directory.  If not found, "
+		"it looks in the home directory, if not found it uses GMT defaults.]", GMT_SETTINGS_FILE);
+	GMT_Usage (API, -2, "Note: Only settings that differ from the GMT SI system defaults are written "
+		"to the file %s in the current directory (under classic mode) "
+		"or in the current session directory (under modern mode).", GMT_SETTINGS_FILE);
+	GMT_Usage (API, 1, "\n-%s", GMT_SHORTHAND_OPTIONS);
```

Comment:
Shorthand options should be -B[JRXYp]<value> rather than -BJRXYp

Reply:
Why is that?  GMT_SHORTHAND_OPTIONS cannot be used in that way.

Reply:
Sorry typo. The syntax and online docs have -[BJRXYp]<value>

Reply:
OK, I see.  I am guessing this is a bit lazy and should be -[B|J|R|X|Y|p]<value> ?

Reply:
> OK, I see. I am guessing this is a bit lazy and should be -[B|J|R|X|Y|p] ?

Yes